docs/develop/go/test-suites.mdx

```diff
@@ -283,6 +286,102 @@ Learn to skip time manually in the SDK of your choice.
 
+## Testing Nexus Operations {#test-nexus}
+
+A Nexus operation can be tested with a mock Nexus environment, which provides a way to mock the
+Nexus operation. This behavior allows you to test the Nexus operation in isolation by calling it
+directly without needing to create a Nexus handler service to run the Nexus operation.
+
+### Mock and override Nexus operations
+
+Let's take a look at a test that simulates a Nexus operation call. Assume that the Nexus operation
+is called `sample-operation`, the input type is `SampleInput`, the output type is `SampleOutput`,
+and it belongs to the Nexus service `sample-service`.
+
+```go
+func (s *UnitTestSuite) Test_SimpleWorkflow_NexusSyncOperation() {
+        s.env.OnNexusOperation(
+                "sample-service",
+                nexus.NewOperationReference[SampleInput, SampleOutput]("sample-operation"),
+                SampleInput{},
+                workflow.NexusOperationOptions{},
+        ).Return(
+                &nexus.HandlerStartOperationResultSync[SampleOutput]{
+                        Value: SampleOutput{},
+                },
+                nil, // error if you want to simulate an error in the ExecuteOperation call
+        )
+        s.env.ExecuteWorkflow(SimpleWorkflow, "test_nexus_operation")
+
+        s.True(s.env.IsWorkflowCompleted())
+        s.NoError(s.env.GetWorkflowError())
+}
+```
+
+The example above simulates a call to a Nexus synchronous operation, indicated by the returned value
+type `*nexus.HandlerStartOperationResultSync[T]`. Since `OnNexusOperation` needs to know the
+operation's name, input and output, `nexus.NewOperationReference` creates an operation reference
+that represents the operation without knowing the implementation (basically, it represents the
+signature of the operation). You may also use the operation itself instead of creating the operation
+reference if you have it available.
+
+Let's take an example of simulating a call to an Nexus asynchronous operation.
+
+```go
+func (s *UnitTestSuite) Test_SimpleWorkflow_NexusAsyncOperation() {
+        s.env.OnNexusOperation(
+                "sample-service",
+                nexus.NewOperationReference[SampleInput, SampleOutput]("sample-operation"),
+                SampleInput{},
+                workflow.NexusOperationOptions{},
+        ).Return(
+                &nexus.HandlerStartOperationResultAsync{
+                        OperationID: "sample-operation-id",
+                },
+                nil, // error if you want to simulate an error in the ExecuteOperation call
+        )
+        err := env.RegisterNexusAsyncOperationCompletion(
+                "sample-service",
+                "sample-operation",
+                "sample-operation-id", // must match the OperationID above
+                SampleOutput{},
+                nil,            // error if you want to simulation an error in the operation
+                2*time.Second,  // delay to simulate how long the operation takes
```

Comment:
This is from the time the operation start was simulated, right? Maybe clarify that?